### PR TITLE
[1497] Add can_add_more_sites? to provider serializer

### PR DESCRIPTION
**Pull request by tvararu**
### Context

We need to hide the button to add new locations when this is false.

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by tvararu** [comment]
[Publishers shouldn't be able to add more than 37 training locations](https://trello.com/c/1HgAUmLX/1497-publishers-shouldnt-be-able-to-add-more-than-37-training-locations)

File: spec/requests/api/v2/providers_spec.rb [hunk 1]
```diff
@@ -236,6 +240,46 @@
...
+      let!(:provider) { create(:provider, course_count: 0, site_count: Site::POSSIBLE_CODES.size, organisations: [organisation]) }
+
+      before do
+        get "/api/v2/providers/#{provider.provider_code}",
+          headers: { 'HTTP_AUTHORIZATION' => credentials }
+      end
+
+      it 'has a data section with the correct attributes' do
+        json_response = JSON.parse(response.body)
+        expect(json_response).to eq(
```
Comment (benilovj) [inline spec/requests/api/v2/providers_spec.rb]: I wonder whether we should only assert the `can_add_more_sites?` on this response, to make the spec more explicit and less prone to breakage if unrelated things change?

Comment (tvararu) [reply spec/requests/api/v2/providers_spec.rb]: Agree

File: spec/models/provider_spec.rb [hunk 2]
```diff
@@ -140,4 +140,16 @@
...
+    it "should return true when provider has less sites than max allowed" do
+      provider = create(:provider, site_count: 0)
+      expect(provider.can_add_more_sites?).to eq(true)
+    end
+
+    it "should return false when provider has the max sites allowed" do
+      provider = create(:provider, site_count: Site::POSSIBLE_CODES.size)
+      expect(provider.can_add_more_sites?).to eq(false)
+    end
+  end
```
Comment (benilovj) [inline spec/models/provider_spec.rb]: I'd have a slight preference for the following style which is what we've been doing in other places:

```ruby
  describe "#can_add_more_sites?" do
    context "when provider has less sites than max allowed" do
      subject { create(:provider, site_count: 0) }
      its(:can_add_more_sites?) { should be_truthy }
    end

    context "when provider has the max sites allowed" do
      subject { create(:provider, site_count: 37) }
      its(:can_add_more_sites?) { should be_falsey }
    end
  end
```

@misaka et al, thoughts?

File: spec/requests/api/v2/providers_spec.rb [hunk 3]
```diff
@@ -236,6 +240,20 @@
       end
     end
 
+    context "with the maximum number of sites" do
+      let!(:provider) { create(:provider, course_count: 0, site_count: Site::POSSIBLE_CODES.size, organisations: [organisation]) }
```
Comment (misaka) [inline spec/requests/api/v2/providers_spec.rb]: Minor thing, but you don't need `let!` here as the `get` request below already references `provider`. It can also get in the way in the future by creating extra data you don't need.

Comment (tvararu) [reply spec/requests/api/v2/providers_spec.rb]: There's a `let!` above and I don't know why

Comment (benilovj) [reply spec/requests/api/v2/providers_spec.rb]: It was probably copied from the spec above that one :trollface:

**Comment by tvararu** [comment]
[Publishers shouldn't be able to add more than 37 training locations](https://trello.com/c/1HgAUmLX/1497-publishers-shouldnt-be-able-to-add-more-than-37-training-locations)
